sessionctx/binloginfo/binloginfo.go

```diff
@@ -109,7 +111,9 @@ func (info *BinlogInfo) WriteBinlog(clusterID uint64) error {
 	// Retry many times because we may raise CRITICAL error here.
 	for i := 0; i < 20; i++ {
 		var resp *binlog.WriteBinlogResp
-		resp, err = info.Client.WriteBinlog(context.Background(), req)
+		ctx, cancel := context.WithTimeout(context.Background(), binlogWriteTimeout)
+		defer cancel()
```

Comment:
We better not defer in a loop, the defer `cancel` is called after the function exit, not the loop.